3.processing_features/scripts/2.pycytominer_singlecell_pipelines.py

```diff
@@ -0,0 +1,157 @@
+pprint.pprint(plate_info_dictionary, indent=4)
+
+
+# ## Perform single-cell pycytominer pipeline
+
+# In[4]:
+
+
+for plate, info in plate_info_dictionary.items():
+    print(f"Now performing single-cell pycytominer pipeline for {plate}")
+    output_annotated_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_annotated.parquet"))
+    output_normalized_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_normalized.parquet"))
+    output_feature_select_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_feature_selected.parquet"))
+    output_feature_select_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_feature_selected.parquet"))
+    output_aggregated_file = str(pathlib.Path(f"{output_dir}/{plate}_bulk_camerons_method.parquet"))
+
+    # Load single-cell profiles
+    single_cell_df = pd.read_parquet(info["dest_path"])
+    
+    # Load platemap
+    platemap_df = pd.read_csv(info["platemap_path"])
+    
+    # Step 1: Annotation
+    # add metadata from platemap file to extracted single cell features
+    annotated_df = annotate(
+        profiles=single_cell_df,
+        platemap=platemap_df,
+        join_on=["Metadata_well_position", "Image_Metadata_Well"],
+    )
+
+    # rename site column to avoid any issues with identifying the column as metadata over feature
+    annotated_df = annotated_df.rename(columns={"Image_Metadata_Site": "Metadata_Site"})
+
+    # move metadata well, single cell count, and site to the front of the df (for easy visualization in python)
+    well_column = annotated_df.pop("Metadata_Well")
+    singlecell_column = annotated_df.pop("Metadata_number_of_singlecells")
+    site_column = annotated_df.pop("Metadata_Site")    
+
+    # insert the columns in specific parts of the dataframe
+    annotated_df.insert(2, "Metadata_Well", well_column)
+    annotated_df.insert(3, "Metadata_Site", site_column)
+    annotated_df.insert(4, "Metadata_number_of_singlecells", singlecell_column)
+
+    # save annotated df as parquet file
+    output(
+        df=annotated_df,
+        output_filename=output_annotated_file,
+        output_type="parquet",
+    )
+    
+    # Step 2: Normalization
+    normalized_df = normalize(
+        profiles=output_annotated_file,
+        method="standardize",
+        output_file=output_normalized_file,
+        output_type="parquet",
+    )
+    
+    # Step 3: Feature selection
+    feature_select(
+        output_normalized_file,
+        operation=feature_select_ops,
+        output_file=output_feature_select_file,
+        output_type="parquet"
+    )
+
+    # Step 4: Cameron's method of aggregation
+    feature_select_df = load_profiles(output_feature_select_file)
+    metadata_cols = infer_cp_features(feature_select_df, metadata=True)
+    metadata_cols = [x for x in metadata_cols if all(col not in x for col in cameron_unwanted_aggregate_cols)]
+    
+    aggregate_df = aggregate(
+        population_df=feature_select_df,
+        operation="median",
+        strata=metadata_cols,
```

Comment:
Same as above comment, wouldn't it save lines of code if this just used the traditional method of ["Metadata_Plate", "Metadata_Well"] as this should be the same as using all of the metadata columns listed above?

Reply:
I believe I answered this question sufficiently in https://github.com/WayScience/nf1_cellpainting_data/pull/23#discussion_r1306079346 but please let me know if not